05/app.js

```diff
@@ -1,53 +1,86 @@
 const stats = {
-    paragraphs: {
-        'p1': 0,
-    },
-    links: {
-        '/dolor.html': 0,
-    }
+	paragraphs: {
+		p1: 0,
+	},
+	links: {
+		'/dolor.html': 0,
+	},
 };
 
 /* tutaj umieść swój kod */
+const pElementsList = document.querySelectorAll('.text');
 
+const countLinkClick = function (e) {
+	const href = e.target.getAttribute('href');
+	if (typeof stats['links'][href] === 'undefined') {
+		stats['links'][href] = 0;
+	}
+	stats['links'][href]++;
+};
+
+const countTextClick = function (e) {
+	const id = e.currentTarget.dataset.id;
+	if (typeof stats['paragraphs'][id] === 'undefined') {
+		stats['paragraphs'][id] = 0;
+	}
+	stats['paragraphs'][id]++;
+};
```

Comment:
Zobacz, że w tych funkcjach masz sporo powielonego kodu.
Może dałoby się napisać dodatkową funkcję, która ten kod zduplikowany usuwa?
Chodzi mi o te cześci:
```
    if (typeof stats['paragraphs'][id] === 'undefined') {
		stats['paragraphs'][id] = 0;
	}
	stats['paragraphs'][id]++;
```
oraz
```
    if (typeof stats['links'][href] === 'undefined') {
		stats['links'][href] = 0;
	}
	stats['links'][href]++;
```

Elementy, który ię różnią to `['paragraphs']` -> `['links']`, `[id]` -> `[href]` - skoro te elementy sie zmieniają to zapewne będą to parametry tej funkcji :)